modules/admission-webhook-deploy.adoc

```diff
@@ -0,0 +1,373 @@
+  scope: Cluster  <4>
+  names:
+    plural: namespacereservations  <5>
+    singular: namespacereservation  <6>
+    kind: NamespaceReservation  <7>
+----
+<1> Reflects Custom Resource Definition `spec` values and is in the format `<plural>.<group>`. This example uses the `namespacereservations` resource.
+<2> REST API group name.
+<3> REST API version name.
+<4> Accepted values are `Namespaced` or `Cluster`.
+<5> Plural name to be included in URL.
+<6> Alias seen in `oc` output.
+<7> The reference for resource manifests.
+
+. Deploy the Custom Resource Definition within the cluster:
++
+----
+$ oc apply -f webhook-crd.yaml
+----
+
+. Configure the webhook server also as an aggregated API server, within a file called `webhook-api-service.yaml`:
++
+[source,yaml]
+----
+apiVersion: apiregistration.k8s.io/v1beta1
+kind: APIService
+metadata:
+  name: v1beta1.admission.online.openshift.io
+spec:
+  caBundle: <ca_signing_certificate>  <1>
+  group: admission.online.openshift.io
+  groupPriorityMinimum: 1000
+  versionPriority: 15
+  service:
+    name: server
+    namespace: my-webhook-namespace
+  version: v1beta1
+----
+<1> A PEM-encoded CA certificate that signs the server certificate that is used by the webhook server. Replace `<ca_signing_certificate>` with the appropriate certificate in base64 format.
+
+. Deploy the aggregated API service:
++
+----
+$ oc apply -f webhook-api-service.yaml
+----
+
+. Define the webhook admission plug-in configuration within a file called `webhook-config.yaml`. This example uses the validating admission plug-in:
++
+[source,yaml]
+----
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: ValidatingWebhookConfiguration
+metadata:
+  name: namespacereservations.admission.online.openshift.io  <1>
+webhooks:
+- name: namespacereservations.admission.online.openshift.io  <2>
+  clientConfig:
+    service:  <3>
+      namespace: default
+      name: kubernetes
+      path: /apis/admission.online.openshift.io/v1beta1/namespacereservations  <4>
+    caBundle: <ca_signing_certificate>  <5>
+  rules:
+  - operations:
+    - CREATE
+    apiGroups:
+    - project.openshift.io
+    apiVersions:
+    - "*"
+    resources:
+    - projectrequests
+  - operations:
+    - CREATE
+    apiGroups:
+    - ""
+    apiVersions:
+    - "*"
+    resources:
+    - namespaces
+  failurePolicy: Fail
+----
+<1> Name for the webhook object. This example uses the `namespacereservations` resource.
+<2> Name of the webhook to call. This example uses the `namespacereservations` resource.
+<3> Enables access to the webhook server through the aggregated API.
+<4> The webhook URL used for admission requests. This example uses the `namespacereservation` resource.
+<5> A PEM-encoded CA certificate that signs the server certificate that is used by the webhook server. Replace `<ca_signing_certificate>` with the appropriate certificate in base64 format.
+
+. Deploy the webhook:
++
+----
+$ oc apply -f webhook-config.yaml
+----
```

Comment:
Consider having a result statement. If you're going for the namespace reservation example, you could give a summary of that - that this example uses the namespace reservations validating webhook to disallow namespace creation if the namespace is reserved. Or if you don't want to get that specific, than just a simple sentence wrap up would be nice to round out this long example procedure. Just something to summarize what they accomplished with these steps.

Reply:
Resolved.